components/brave_wallet_ui/common/wallet_api_proxy_observers.ts

```diff
@@ -192,14 +191,11 @@ export function makeJsonRpcServiceObserver (store: Store) {
 
   export function  makeBraveWalletPinServiceObserver (store: Store) {
     const braveWalletServiceObserverReceiver = new BraveWallet.BraveWalletPinServiceObserverReceiver({
-      onTokenStatusChanged: function (service, token, status) {
-        store.dispatch(WalletPageActions.updateNftPinningStatus({
-          token,
-          status
-        }))
+      onTokenStatusChanged: function (_service, _token, _status) {
+        store.dispatch(walletApi.endpoints.getNftsPinningStatus.initiate())
```

Comment:
Are we sure we want to refresh the status for ALL visible tokens when this event is fired? If not, we should add a mutation that performs an ["optimistic update"](https://redux-toolkit.js.org/rtk-query/usage/manual-cache-updates#optimistic-updates) to the state of the `getNftsPinningStatus` query cache.